xlator/options.go

```diff
@@ -65,3 +71,176 @@ type Option struct {
 	Tags         []string
 	SetKey       string
 }
+
+// ValidateFunc calls the respective validate function
+func (o *Option) ValidateFunc(val string) error {
+	var err error
+	switch t := o.Type; t {
+	case OptionTypeBool:
+		err = ValidateBool(o, val)
+	case OptionTypeClientAuthAddr:
+		err = ValidateClientAuthAddr(o, val)
+	case OptionTypeDouble:
+		err = ValidateDouble(o, val)
+	case OptionTypeInt:
+		err = ValidateInt(o, val)
+	case OptionTypeInternetAddress:
+		err = ValidateInternetAddress(o, val)
+	case OptionTypeInternetAddressList:
+		err = ValidateInternetAddressList(o, val)
+	case OptionTypePath:
+		err = ValidatePath(o, val)
+	case OptionTypePercent:
+		err = ValidatePercent(o, val)
+	case OptionTypePercentOrSizet:
+		err = ValidatePercentOrSize(o, val)
+	case OptionTypePriorityList:
+		err = ValidatePriorityOrSize(o, val)
+	case OptionTypeSizeList:
+		err = ValidateSizeList(o, val)
+	case OptionTypeSizet:
+		err = ValidateSizet(o, val)
+	case OptionTypeStr:
+		err = ValidateStr(o, val)
+	case OptionTypeTime:
+		err = ValidateTime(o, val)
+	case OptionTypeXlator:
+		err = ValidateXlator(o, val)
+	default:
+		err = ValidateOption(o, val)
+	}
+	return err
+}
+
+// ValidateBool validates if the option is of type boolean
+func ValidateBool(o *Option, val string) error {
+	if val == "" {
+		err := errors.New("No argument passed")
```

Comment:
It is better to define common error types for common validation errors, instead of creating new errors every time.